src/DS/metrics_extraction.py

```diff
@@ -0,0 +1,300 @@
+"""
+This script analyses Python test files within cloned repositories to extract
+relevant metrics and perform various analyses. The script processes both test
+files and general Python code files, extracting metrics such as:
+
+- Number of test cases (functions starting with 'test_')
+- Number of assertions (using 'assert' statements)
+- Presence of setup and teardown methods (e.g., setUp, tearDown)
+- Complexity of the test file (measured by counting functions and branches)
+- Cyclomatic complexity of the code : sum of all functions' complexities where
+cyclomatic complexity of that function, is a measure of how many independent
+paths or 'decision points' exist in the function. Cyclomatic complexity
+generally increases with the number of branches, loops, and conditional
+statements in the function.
+- Lines of code (LOC) : Counts lines by splitting the file content into lines.
+- Number of functions : Counts each function encountered.
+
+The script saves the results of these analyses to a specified CSV file,
+updating the file in batches as multiple files are processed.
+
+### Usage:
+To run the script, specify the input and output CSV file paths using the
+command-line arguments `--input` and `--output`. By default, the script
+assumes input and output files are located in the 'data' directory of the
+repository's root.
+
+Example:
+    python script_name.py --input path/to/input.csv --output path/to/output.csv
+
+"""
+
+import argparse
+import ast
+import pandas as pd
+from mccabe import PathGraphingAstVisitor
+from loguru import logger
+from pathlib import Path
+
+from utils.git_utils import get_working_directory_or_git_root
+
+BATCH_SIZE = 100  # Number of files to process before saving to disk
+
+# Configure logger
+logger.add("metrics_extraction_log", rotation="500 MB", level="INFO")
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Analyse Python test files and extract metrics."
```

Comment:
Minor correction, in lines 3 and 4 of this file you've written: `The script processes both test files and general Python code files` while this only mentions test files. Please can you revise these to be consistent?